misc/git/hooks/govet

```diff
@@ -24,25 +24,14 @@ if [ -z "$GOPATH" ]; then
 fi
 
 # This script does not handle file names that contain spaces.
-gofiles=$(git diff --cached --name-only --diff-filter=ACM | grep '^go/.*\.go$')
+packages=$(git diff --cached --name-only --diff-filter=ACM | grep '^go/.*\.go$' | sed 's|/[^/]*$|/...|' | uniq | sed 's|^|./|')
```

Comment:
I tried this locally and it returns something like `./go/vt/vttablet/tabletmanager/...` and `go vet` barfs on that.
Why didn't you use what we have in `staticcheck`?
`gopackages=$(echo $gofiles | xargs -n1 dirname | sort -u)`

Reply:
i didn't see the staticcheck thing, but that doesn't work for me:

Reply:
`dirname: missing operand`

Reply:
what platform are you on?

Reply:
ubuntu

Reply:
`go vet` needs to be invoked on `vitess.io/vitess/$gopackage` rather than directly on `$gopackages`.
When I tried this locally, go an error:
```can't load package: package go/vt/wrangler/testlib: cannot find package "go/vt/wrangler/testlib" in any of:
	/usr/local/go/src/go/vt/wrangler/testlib (from $GOROOT)
	/home/deepthi/go/src/go/vt/wrangler/testlib (from $GOPATH)

Please fix the go vet warnings above. To disable certain checks, change vetflags in misc/git/hooks/govet.
```

Reply:
Consider rewriting the govet hook to follow the same pattern as the staticcheck hook (starting with #!bin/bash) because we know that works.